src/tests/suite/daos_epoch_io.c

```diff
@@ -267,7 +267,8 @@ fio_test_cb_uf(test_arg_t *arg, struct test_op_record *op, char **rbuf,
 		D_GOTO(out, rc = -DER_NOMEM);
 	if (op->or_op == TEST_OP_UPDATE)
 		test_buf_init(buf, buf_size, uf_arg->ua_recxs,
-			      uf_arg->ua_values, uf_arg->ua_recx_num,
+			      uf_arg->ua_values ?: &uf_arg->ua_single_value,
```

Comment:
TIL...never seen that trick before.   I might err on the side of being explicit though to make the code more clear to the future reader given my first reaction was, "what does that do?"

Reply:
I think that is just to initialize the data with specific value, that change is to fix a segfault I met.

Reply:
Oh, I was referring to the ?: part.  I'd never seen that syntax used so had to lookup what it does.

x :? y is equivalent to x ? x : y

I think it's clearer to the reader if it's explicitly duplicated as this is a somewhat obscure C extension.

uf_arg->ua_values ? uf_arg->ua_values : &uf_arg->ua_single_value

But not a big deal.